static/js/upload.js

```diff
@@ -13,187 +17,223 @@ exports.feature_check = function (upload_button) {
+        return;
+    }
+    if (page_params.max_file_upload_size === 0) {
+        exports.show_error_message(config, i18n.t('File and image uploads have been disabled for this organization.'));
+        return;
+    }
+    exports.get_item("send_button", config).attr("disabled", "");
+    exports.get_item("send_status", config).addClass("alert-info").removeClass("alert-error").show();
+    exports.get_item("send_status_message", config).html($("<p>").text(i18n.t("Uploading…")));
+    exports.get_item("send_status_close_button", config).one('click', function () {
+        uppy.cancelAll();
+        setTimeout(function () {
+            exports.hide_upload_status(config);
+        }, 500);
+    });
+
+    for (const file of files) {
+        try {
+            compose_ui.insert_syntax_and_focus("[Uploading " + file.name + "…]()", exports.get_item("textarea", config));
+            compose_ui.autosize_textarea();
+            uppy.addFile({
+                source: exports.get_item("source", config),
+                name: file.name,
+                type: file.type,
+                data: file,
+            });
+        } catch (error) {
+            // Errors are handled by info-visible and upload-error event callbacks.
         }
+    }
+};
 
-        switch (error_code) {
-        case 'BrowserNotSupported':
-            msg = i18n.t("File upload is not yet available for your browser.");
-            break;
-        case 'TooManyFiles':
-            msg = i18n.t("Unable to upload that many files at once.");
-            break;
-        case 'FileTooLarge':
-            if (page_params.max_file_upload_size > 0) {
-                // sanitization not needed as the file name is not potentially parsed as HTML, etc.
-                const context = {
-                    file_name: file.name,
-                    file_size: page_params.max_file_upload_size,
-                };
-                msg = i18n.t('"__file_name__" was too large; the maximum file size is __file_size__MB.',
-                             context);
-            } else {
-                // If uploading files has been disabled.
-                msg = i18n.t('File and image uploads have been disabled for this organization.');
-            }
-            break;
-        case 413: // HTTP status "Request Entity Too Large"
-            msg = i18n.t("Sorry, the file was too large.");
-            break;
-        case 400: {
-            const server_message = server_response && server_response.msg;
-            msg = server_message || i18n.t("An unknown error occurred.");
-            break;
+exports.setup_upload = function (config) {
+    const uppy = Uppy({
+        debug: true,
+        autoProceed: true,
+        restrictions: {
+            maxFileSize: page_params.max_file_upload_size * 1024 * 1024,
+        },
+        locale: {
+            strings: {
+                exceedsSize: i18n.t('This file exceeds maximum allowed size of'),
```

Comment:
This one it seems like we should definitely be able to just do the i18n substitution here, since the max file upload size is immutable.

Reply:
Actually Uppy takes care of substituting the value of max file size.

https://github.com/transloadit/uppy/blob/master/packages/%40uppy/core/src/index.js#L460

I can imagine this resulting in not grammatically correct sentences in some languages. But I think we don't have much options here.

Reply:
Yeah, it looks like their i18n here is fundamentally broken on this front.  I opened https://github.com/transloadit/uppy/issues/2069 to track this issue upstream; it seems like it's probably not a super hard fix aside from the fact that it requires them to change their interface.